### ISSUE TITLE
Add FNet as a optional implementation of ViT

### ISSUE THREAD

**Issue by FilipAndersson245** [lucidrains/vit-pytorch#123]
[Arxiv](https://arxiv.org/abs/2105.03824), [Yannic](https://www.youtube.com/watch?v=JJR3pBl78zw)
The authors proposes that you can replace attention with Fourier transformations on BERT, this improves speed immensely (~6x) with just minor loss in predictive performance.
maybe it would be interesting to examine if it can be integrated in ViT.

**Comment by lessw2020** [comment]
I think this is also quite interesting.  I'd recommend building the hybrid model implementation for more optimal speed/accuracy tradeoff:

> "we found that adding self-attention sublayers to FNet models offers a simple way to trade off speed for accuracy...
specifically replacing the final two Fourier sublayers of FNet with self-attention layers yielded a model that acheived 97% of BERT accuracy, but pre-trained six times as fast on gpus..."

And go with NesT transformer as the home instead of ViT :)
